frontend/src/routes/wallet/keypair/taint.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState, useCallback } from 'react'
+import { Intent } from '../../../config/intent'
+import { Header } from '../../../components/header'
+import { Button } from '../../../components/button'
+import { AppToaster } from '../../../components/toaster'
+import { BreakText } from '../../../components/break-text'
+import { KeyValueTable } from '../../../components/key-value-table'
+import { useCurrentKeypair } from '../../../hooks/use-current-keypair'
+import { requestPassphrase } from '../../../components/passphrase-modal'
+import { updateKeyPairAction } from '../../../contexts/global/global-actions'
+import { useGlobal, GlobalDispatch } from '../../../contexts/global/global-context'
+import { createLogger } from '../../../lib/logging'
+import { Service } from '../../../service'
+
+const logger = createLogger('Taint')
+
+const useTaint = (dispatch: GlobalDispatch, pubKey?: string, wallet?: string) => {
+  const [loading, setLoading] = useState(false);
+
+  const taint = useCallback(
+    async () => {
+      setLoading(true);
+      try {
+        if (!pubKey || !wallet) {
+          return
+        }
+
+        const passphrase = await requestPassphrase()
+        await Service.TaintKey({
+          wallet,
+          pubKey,
+          passphrase
+        });
+
+        const keypair = await Service.DescribeKey({
+          wallet,
+          pubKey,
+          passphrase
+        });
+
+        dispatch(updateKeyPairAction(wallet, keypair));
+
+        setLoading(false);
+        AppToaster.show({
+          message: `This key has been tainted`,
+          intent: Intent.SUCCESS
+        })
+      } catch (err) {
+        setLoading(false);
+        AppToaster.show({ message: `${err}`, intent: Intent.DANGER })
+        logger.error(err)
+      }
+    },
+    [pubKey, wallet]
+  )
+
+  const untaint = useCallback(
```

Comment:
question (blocking): Do we not need to call `dispatch(updateKeyPairAction(wallet,keypair)` when untainting?

Reply:
yes, totally. added now